app/assets/javascripts/discourse/app/routes/build-category-route.js

```diff
@@ -29,9 +30,11 @@ class AbstractCategoryRoute extends DiscourseRoute {
   controllerName = "discovery/list";
 
   async model(params, transition) {
-    const category = Category.findBySlugPathWithID(
-      params.category_slug_path_with_id
-    );
+    const category = this.siteSettings.lazy_load_categories
+      ? await Category.asyncFindBySlugPathWithID(
+          params.category_slug_path_with_id
+        )
```

Comment:
In the context of this PR, what problem is this solving? Also, I wonder if we should first check if the category has already been preloaded instead of triggering a network request to fetch the categories unconditionally.

Reply:
Yes and no. I mean it is not a direct fix to category-drop, but navigating using category-drop is broken otherwise.